tests/fsharp/HandleExpects.fs

```diff
@@ -0,0 +1,196 @@
+module HandleExpects
+
+open System
+open System.IO
+open System.Text.RegularExpressions
+open System.Xml
+
+type Expects = { status:string; id:string; span:string; pattern:string; mutable matched:bool; line:string }
+type ErrorMessage = { source:string; status:string; id:string; span:string; text:string; mutable matched:bool; line:string }
+type Span = { startrow:int; startcol:int; endrow:int; endcol:int }
+
+let tryParseSpan (span:string) =
+    let s = span.Trim([| '('; ')' |]).Split(',')
+    match s.Length with
+    | 2 -> { startrow=Int32.Parse(s.[0]); startcol=Int32.Parse(s.[1]); endrow=Int32.MaxValue; endcol=Int32.MaxValue }
+    | 4 -> { startrow=Int32.Parse(s.[0]); startcol=Int32.Parse(s.[1]); endrow=Int32.Parse(s.[2]); endcol=Int32.Parse(s.[3]) }
+    | _ -> raise (InvalidDataException(sprintf "The span : '%s' is invalid" span));
+
+let isStringEmpty s = String.IsNullOrWhiteSpace(s)
+let isStringNotEmpty s = not (isStringEmpty s)
+let stringToLower s = if isStringNotEmpty s then s.ToLower() else s
+let areStringsEqual s1 s2 = String.Compare(s1, s2, StringComparison.OrdinalIgnoreCase) = 0 
+let areSpansEqual s1 s2 =
+    let span1 = tryParseSpan s1
+    let span2 = tryParseSpan s2
+    if span1.startrow <> span2.startrow then false
+    elif span1.startcol <> span2.startcol then false
+    elif span1.endrow <> span2.endrow then false
+    elif span1.endcol <> span2.endcol then false
+    else true
+
+let stripFromFileExpectations source =
+    let readExpect expect =
+        let pattern = "(?<tagOpen><Expects{1}[^>]*>{1})(?<tagContent>.*)(?<tagClose></Expects>)"          //"(<!--((?!-->).)*-->|<\w*((?!\/<).)*\/>|<(?<Expects>\w+)[^>]*>(?>[^<]|(?R))*<\/\k<Expects>\s*>)"
+        let rx = new Regex(pattern)
+        let matched = rx.Match(expect)
+        if matched.Success then
+            // The content of the Expects group contains a lot of invalid Xml and the Xml reader fails when it sees it.
+            // So we just save it away, remove it from the xml, then read the xml and put it back
+            // Save away the contents of the element and strip it out of expect pattern
+            let content = (matched.Groups.[2]).ToString()
+            let nocontentxpect = 
+                if isStringEmpty content then expect
+                else expect.Replace(content, "")
+
+            let rdr = XmlReader.Create(new StringReader(nocontentxpect))
```

Comment:
Why not `System.Xml.Linq.XDocument`?  It's (I think) a much better API than using a reader.

Reply:
If I had used Google search for samples I'm sure I would have got the good api, but I used Bing.